app/utils/model-save.js

```diff
@@ -1,46 +1,120 @@
 import { isNone } from '@ember/utils';
-
 export default class ModelSaveUtil {
   constructor(entity) {
     this.entity = entity;
   }
 
-  onSuccess(model) {
+  sendSuccess() {
+    if (!isNone(this.entity?.successMessage)) {
+      this.entity?.flashNotice?.sendSuccess(this.entity.successMessage);
```

Comment:
If above condition passes, entity certainly exists here so the chaining is not necessary. Happens a few occasions down here too

Reply:
very true, I was just being cautious. I'll clean that up a bit if you'd like me to?

Reply:
fixed all cases I could find in modelSaveUtils